### PR TITLE
[FOR REVIEW] Implement methods for Company resource

### PR DESCRIPTION
**Pull request by wnuqui**
Reference: https://reference.assemblypayments.com/#companies

## Todos
- [x] [Create Company](https://reference.assemblypayments.com/#create-company)
- [x] [List Companies](https://reference.assemblypayments.com/#list-companies)
- [x] [Show Company](https://reference.assemblypayments.com/#show-company)
- [x] [Update Company](https://reference.assemblypayments.com/#update-company)
- [x] Add tests
- [x] Update API documentation

### CODE REVIEW

**Review by psyfear — APPROVED** [review]

**Comment by wnuqui** [comment]
@psyfear look! The coverage is going down. 😛 Later will add tests.

**Comment by wnuqui** [comment]
Ready for review/merge. Thanks!

File: test/promisepay_ex/companies/companies_test.exs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  setup do
+    PromisepayEx.configure(
+      username: "test.test@test.com",
+      password: "test",
+      environment: "test",
+      api_domain: "api.localhost.local:3000",
+    )
+
+    Config.filter_request_headers("Authorization")
```
Comment (psyfear) [inline test/promisepay_ex/companies/companies_test.exs:27]: do we still need to do this? we're authenticating via username/password now

**Review by psyfear — APPROVED** [review]

File: test/promisepay_ex/companies/companies_test.exs (hunk 1 above)
Comment (wnuqui) [reply test/promisepay_ex/companies/companies_test.exs:27]: Nice catch!

**Comment by psyfear** [comment]
putting the vcr fixtures inside object/METHOD/HTTP_CODE.json is a pretty good idea. There are some resources that aren't restful though. What would you suggest for them?

File: test/promisepay_ex/companies/companies_test.exs (hunk 1 above)
Comment (wnuqui) [reply test/promisepay_ex/companies/companies_test.exs:27]: An update here: https://github.com/psyfear/promisepay_ex/pull/18 . Please review/merge. Thanks!

**Comment by wnuqui** [comment]
Yup, had revisions before finally ending up with `[resource]/[method]/[http_code].json` format. As for non-restful resource, I actually do not have idea for now.

**Comment by psyfear** [comment]
thanks @wnuqui and good job!

**Pull request merged**